### PR TITLE
E2E Test Setup & Teardown Refactor

### PR DESCRIPTION
**Pull request by patthiel**
* This is a significant refactor that changes the design pattern used for creating e2e tests. The idea is that now, all setup/teardown code should be baked into a test spec file. 

* To combat unexpected failures resulting in an unexpected account state after a test run, the setup/teardown utils use the v4 API to create / delete things like linodes, domains, volumes, etc. 
* Also used this PR as an opportunity to fix issues plaguing the docker e2e. We now call `yarn build && yarn serve` to first build the static assets then start a simple https server to run the manager app. This prevents docker container start timing issues.

## Setup/Teardown Related Changes:

* Updated `.env.example` to warn contributors about running the e2e tests
* Added custom command for `readToken` to get the token value from `./localStorage.json`
* Added custom commands that call api create/delete utilities. Needed because any async code used by `wdio` test runner has to be wrapped in a custom command like so: `browser.addCommand('someAsyncCommand', function async() {`
* Fixed issues with existing setup/teardown method's error logging 
* Added `apiCreateLinode` , `apiDeleteAllLinodes` and `apiDeleteAllVolumes` utilities
* Added command line flags to wdio that will set the baseUrl.. `--dev`, `--test`, `--staging`, `--prod`
* Removed manual "create linode" step of `setup.spec.js` script.. This script's purpose is to now only remove existing account data. 
* Updated various tests to use api create/delete utilities. 

## Docker Related Changes:
* Added `serve` dependency. one can now run `yarn build` followed by `yarn serve` to start manager and serve static files (Needed to fix timing issues on docker).
* Added docker entrypoint script for manager
* Updated Dockerfile image to latests node/debian version.

## Misc Fixes
* Fixed clone linode tests (no longer has set target handling)
* Fixed search bar tests

# To Test
Main thing we want here is to prove that creating/deleting linodes/volumes works in our tests

0. Make sure your `.env` file is setup using your dev test account credentials
1. `yarn && yarn start`
2. `yarn selenium` (in a new shell)
3. `yarn e2e` (in a new shell)

* I ran the tests yesterdayt and got the following results:
https://gist.github.com/patthiel/51724c061d431a5266cf443d4a272b7b

111 passing (1173.50s)
11 skipped
36 failing

I believe in this test run there were some timeouts which resulted in whole spec files failing all tests..

I ran the tests again this morning and got different results:

133 passing (1026.90s)
11 skipped
14 failing

https://gist.github.com/patthiel/3602604570f793b6e0e8255b26b722c3

* Some of these failures appear to be due to outdated tests or `data-qa` attributes that have been removed. I will address these in a followup PR, as this one is already huge. 

# To Test the Static Server:

```bash
yarn # to install serve
yarn build # build dist files
yarn serve # should start a https server (https is needed for docker tests)
```

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -7,6 +7,8 @@ REACT_APP_API_ROOT='https://api.linode.com/v4'
 #REACT_APP_APP_ROOT='http://localhost:3000'
 
 ## UNCOMMENT AND UPDATE FOR RUNNING E2E TESTS
+## WARNING***** RUNNING E2E TESTS WILL REMOVE ALL DATA
```
Comment (asauber) [inline .env.example]: I would make this warning even stronger. "WILL REMOVE ALL OF YOUR DATA AND RESOURCES INCLUDING LINODES AND VOLUMES"

File: e2e/config/wdio.conf.js [hunk 2]
```diff
@@ -8,10 +8,29 @@ const {
 const { removeAllLinodes } = require('../setup/setup');
 const { browserCommands } = require('./custom-commands');
 const { browserConf } = require('./browser-config');
+const { constants } = require('../constants');
 const selectedBrowser = argv.b ? browserConf[argv.b] : browserConf['chrome'];
 const username = process.env.MANAGER_USER;
 const password = process.env.MANAGER_PASS;
 
+const setBaseUrl = () => {
```
Comment (asauber) [inline e2e/config/wdio.conf.js]: These hostnames should not appear in the codebase, please use the .env file

**Comment by asauber** [comment]
During a test run I got the following error:

```
Error: A session id is required for this command but wasn't found in the response payload
    at new RuntimeError (node_modules/webdriverio/build/lib/utils/ErrorHandler.js:144:12)
    at RequestHandler.createOptions (node_modules/webdriverio/build/lib/utils/RequestHandler.js:121:23)
    at RequestHandler.create (node_modules/webdriverio/build/lib/utils/RequestHandler.js:212:43)
    at Object.elements (node_modules/webdriverio/build/lib/protocol/elements.js:78:32)
    at Object.exec (node_modules/webdriverio/build/lib/helpers/safeExecute.js:28:24)
    at Object.resolve (node_modules/webdriverio/build/lib/webdriverio.js:191:29)
    at lastPromise.then.resolve.call.depth (node_modules/webdriverio/build/lib/webdriverio.js:486:32)
    at _fulfilled (node_modules/q/q.js:854:54)
    at self.promiseDispatch.done (node_modules/q/q.js:883:30)
    at Promise.promise.promiseDispatch (node_modules/q/q.js:816:13)
```

**Comment by asauber** [comment]
Just did a full run, results here https://gist.github.com/asauber/885eeb9c71cd15f8f642c1afdb0464d7

File: e2e/config/wdio.conf.js (hunk 2 above)
Comment (patthiel) [reply e2e/config/wdio.conf.js]: good catch, fixing.

**Comment by patthiel** [comment]
@asauber hmm i've run the suite several times and i have not gotten the session id error. That can happen when a browser session gets removed / exited before a command is run. This can happen if, for instance, you accidentally quit chrome while a test is running. I will keep an eye out for this when running the full suite. Also, noted your errors, I have a separate PR addressing many of those issues.

File: integration-test.yml [hunk 3]
```diff
@@ -36,6 +37,6 @@ services:
     volumes:
       - ./src:/src/src
       - ./e2e:/src/e2e
-    entrypoint: ["./scripts/wait-for-it.sh", "-t", "250", "-s", "manager-local:3000", "--", "yarn","e2e:all", "--log"]
+    entrypoint: ["./scripts/wait-for-it.sh", "-t", "250", "-s", "manager-local:3000", "--", "yarn","e2e", "--log"]
```
Comment (patthiel) [inline integration-test.yml]: updated so that it runs only the e2e tests, `e2e:all` was attempting to run the storybook tests as well

**Pull request merged**